src/basic/io-util.c

```diff
@@ -159,21 +158,29 @@ int pipe_eof(int fd) {
         return !!(r & POLLHUP);
 }
 
-int fd_wait_for_event(int fd, int event, usec_t t) {
+int ppoll_usec(struct pollfd *fds, size_t nfds, usec_t timeout) {
+        struct timespec ts;
+        int r;
+
+        assert(fds);
```

Comment:
hmm, maybe `assert(fds || n_fds == 0)` here? it might make sense to call this with fds after all in some cases